model_utils/tracker.py

```diff
@@ -86,11 +86,46 @@ def __delete__(self, obj):
         self.descriptor.__delete__(obj)
 
 
+class FieldsContext:
+    def __init__(self, tracker, *fields, state=None):
```

Comment:
Maybe you could add a docstring to explain the reason for this second Context Manager. I know this is to handle several states and it is clear when looking at the whole PR now, but might be less obvious when looking at the class inside the code later on.

Reply:
Added a docstring.